### PR TITLE
scim support

### PR DESCRIPTION
**Pull request by jakcinmarina**
- add support for SCIM API
- syncs IDP groups
- provision of IDP group members

### CODE REVIEW

File: pkg/slack/client.go [hunk 1]
```diff
@@ -284,6 +335,111 @@ func (c *Client) SetWorkspaceRole(ctx context.Context, teamID, userID, roleID st
...
+	Value []UserID `json:"value"`
+}
+
+type UserID struct {
+	Value string `json:"value"`
+}
+
+// GroupPatch patches a group by adding or removing a user.
+func (c *Client) GroupPatch(ctx context.Context, groupID string, user string, remove bool) error {
+	groupSCIMUrl, err := url.JoinPath(baseScimUrl, "Groups", groupID)
```
Comment (ggreer) [inline pkg/slack/client.go:406]: Is this url case sensitive? If not, it'd be nice to keep it lowercase.

File: pkg/slack/client.go [hunk 2]
```diff
@@ -302,15 +464,32 @@ func (c *Client) doRequest(ctx context.Context, url string, res interface{}, met
...
 	if err != nil {
 		return err
 	}
 
 	defer resp.Body.Close()
 
-	if err := json.NewDecoder(resp.Body).Decode(&res); err != nil {
-		return err
+	// even though it's stated in the docs that PATCH request should return a resource
+	// for some reason when adding user in IDP group response is 204, but when removing a user it returns Group resource
```
Comment (ggreer) [inline pkg/slack/client.go:486]: 😧  Why, Slack? Why?!

File: pkg/slack/client.go [hunk 3]
```diff
@@ -302,15 +464,32 @@ func (c *Client) doRequest(ctx context.Context, url string, res interface{}, met
 
 	req.Header.Add("Accept", "application/json")
 	req.Header.Set("Content-Type", "application/x-www-form-urlencoded")
+
+	// different headers for SCIM API
+	if strings.Contains(url, baseScimUrl) {
```
Comment (ggreer) [inline pkg/slack/client.go:469]: I know it'll probably never matter, but can we use HasPrefix instead of Contains?

**Review by ggreer — COMMENTED** [review]
Just noticed a few minor things. Besides that it looks good to me.

File: pkg/slack/client.go (hunk 1 above)
Comment (jakcinmarina) [reply pkg/slack/client.go:406]: unfortunately, it is, just tested it

File: pkg/slack/client.go (hunk 2 above)
Comment (jakcinmarina) [reply pkg/slack/client.go:486]: my thoughts exactly 😄

**Review by ggreer — APPROVED** [review]

**Pull request merged**